build_tools/buildkite/dana/upload_benchmarks.py

```diff
@@ -0,0 +1,297 @@
+                 cwd=os.path.dirname(os.path.realpath(__file__)),
+                 capture_output=True,
+                 verbose=verbose).stdout.strip()
+
+
+def get_git_total_commit_count(commit, verbose=False):
+  """Gets the total commit count in history ending with the given commit."""
+  count = execute(['git', 'rev-list', '--count', commit],
+                  cwd=os.path.dirname(os.path.realpath(__file__)),
+                  capture_output=True,
+                  verbose=verbose).stdout.strip()
+  return int(count)
+
+
+def get_git_commit_info(commit, verbose=False):
+  """Gets commit information dictory for the given commit."""
+  info = execute([
+      'git', 'show', '--format=%H:::%h:::%an:::%ae:::%s', '--no-patch', commit
+  ],
+                 cwd=os.path.dirname(os.path.realpath(__file__)),
+                 capture_output=True,
+                 verbose=verbose).stdout.strip()
+  segments = info.split(':::')
+  return {
+      'hash': segments[0],
+      'abbrevHash': segments[1],
+      'authorName': segments[2],
+      'authorEmail': segments[3],
+      'subject': segments[4],
+  }
+
+
+def compose_serie_payload(project_id,
+                          serie_id,
+                          serie_description=None,
+                          average_range='5%',
+                          average_min_count=3,
+                          better_criterion='smaller',
+                          override=False):
+  """Composes the payload dictionary for a serie."""
+  payload = {
+      'projectId': project_id,
+      'serieId': serie_id,
+      'analyse': {
+          'benchmark': {
+              'range': average_range,
+              'required': average_min_count,
+              'trend': better_criterion,
+          }
+      },
+      'override': override,
+  }
+  if serie_description is not None:
+    payload['description'] = serie_description
+  return payload
+
+
+def compose_build_payload(project_id,
+                          project_github_comit_url,
+                          build_id,
+                          commit,
+                          override=False):
+  """Composes the payload dictionary for a build."""
+  commit_info = get_git_commit_info(commit)
+  commit_info['url'] = f'{project_github_comit_url}/{commit_info["hash"]}'
+  return {
+      'projectId': project_id,
+      'build': {
+          'buildId': build_id,
+          'infos': commit_info,
+      },
+      'override': override,
+  }
+
+
+def compose_sample_payload(project_id,
+                           serie_id,
+                           build_id,
+                           sample_value,
+                           override=False):
+  """Composes the payload dictionary for a sample."""
+  return {
+      'projectId': project_id,
+      'serieId': serie_id,
+      'sample': {
+          'buildId': build_id,
+          'value': sample_value
+      },
+      'override': override
+  }
+
+
+def get_env_var(var):
```

Comment:
Maybe `get_required_env_var` to make it clear why this exists?

```suggestion
def get_required_env_var(var):
```